crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -566,10 +569,13 @@ impl<'a> SimpleTokenizer<'a> {
                 let range = TextRange::at(self.offset, token_len);
                 let kind = to_keyword_or_other(&self.source[range]);
 
-                if kind == SimpleTokenKind::Other {
+                // If the next character is a quote, we're in a string prefix. For example: `f"foo`.
+                if kind == SimpleTokenKind::Name && matches!(self.cursor.first(), '"' | '\'') {
```

Comment:
This is an over-approximation and might be fine... but it will return a different output than `Lexer` gives for 

```python
abc"test"
```

Our regular lexer returns `Name` and `String` but this implementation returns `Other`, `Other`. 

It would be nice if we could align the behavior by either matching on all possible valid prefixes OR do the prefix check before lexing the identifier, similar to what `Lexer` does

Reply:
I changed it to be more defensive.